aggregation/simple/src/main/java/com/spotify/heroic/aggregation/simple/DeltaInstance.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (c) 2015 Spotify AB.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.heroic.aggregation.simple;
+import com.spotify.heroic.aggregation.AggregationInstance;
+import com.spotify.heroic.aggregation.AggregationResult;
+import com.spotify.heroic.aggregation.AggregationSession;
+import com.spotify.heroic.aggregation.AggregationOutput;
+import com.spotify.heroic.aggregation.EmptyInstance;
+import com.spotify.heroic.common.DateRange;
+import com.spotify.heroic.common.Series;
+import com.spotify.heroic.metric.*;
+import com.spotify.heroic.metric.Spread;
+import lombok.Data;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Data
+public class DeltaInstance implements AggregationInstance {
+    private final EmptyInstance inner = new EmptyInstance();
+
+    @Override
+    public long estimate(DateRange range) {
+        return -1;
+    }
+
+    @Override
+    public long cadence() {
+        return -1;
+    }
+
+    @Override
+    public AggregationInstance distributed() {
+        return this;
+    }
+
+    @Override
+    public boolean distributable() {
+        return false;
+    }
+
+    public List<Point> computeDiff(List<Point> points) {
+        final List<Point> result = new ArrayList<Point>();
+        double lastValue = points.get(0).getValue();
+        for (int i = 0; i < points.size(); i++) {
+            Point thisPoint = points.get(i);
```

Comment:
In general there's no guarantee that a list implementation has an efficient `get(int)` method, see [LinkedList](https://docs.oracle.com/javase/8/docs/api/java/util/LinkedList.html) for an example. Use an iterator (preferred) or a foreach.

If you want inspiration I usually do the following when I have a previous value that I want to use in the operation:
```java
final Iterator<Point> it = points.iterator();

if (!it.hasNext()) {
    return Collections.emptyList();
}

Point previous = it.next();

while (it.hasNext()) {
    final Point current = it.next();
    /* do your thing */
    previous = current;
}
```